File: luci-app-clashoo/htdocs/luci-static/resources/view/clash/system.js
'use strict';
'require view';
'require form';
'require rpc';
'require uci';
'require poll';
'require tools.clash as clash';

let callGetArch = rpc.declare({ object: 'luci.clash', method: 'get_cpu_arch', expect: { arch: '' } });


return view.extend({
    load: function () {
        return Promise.all([
            L.resolveDefault(callGetArch(), ''),
            uci.load('clash'),
            clash.readLog(),
            clash.readUpdateLog(),
            clash.readGeoipLog()
        ]);
    },

    render: function (data) {
        let cpuArch = (typeof data[0] === 'string' ? data[0] : (data[0]?.arch || '')).trim();
        let logContent    = data[2] || '';
        let updateContent = data[3] || '';
        let geoipContent  = data[4] || '';
        let m, s, o;

        m = new form.Map('clash', '');

        /* ─── 内核下载 ─── */
        s = m.section(form.NamedSection, 'config', 'clash', _('内核下载'));
        s.description = _('从 GitHub 下载 Mihomo 内核二进制文件');
        s.anonymous = false;

        o = s.option(form.ListValue, 'dcore', _('版本类型'));
        o.value('2', 'mihomo（稳定版）');
        o.value('3', 'Alpha（预发布版）');
        o.default = '3';

        o = s.option(form.ListValue, 'download_core', _('CPU 架构'));
        o.value('aarch64_cortex-a53');
        o.value('aarch64_generic');
        o.value('arm_cortex-a7_neon-vfpv4');
        o.value('mipsel_24kc');
        o.value('mips_24kc');
        o.value('x86_64');
        o.value('riscv64');
        o.default = cpuArch || 'x86_64';
        o.description = cpuArch ? _('当前设备：<strong>%s</strong>').format(cpuArch) : '';

        o = s.option(form.Value, 'core_mirror_prefix', _('下载镜像前缀（可选）'));
        o.placeholder = 'https://gh-proxy.com/';
        o.rmempty = true;
        o.description = _('留空直连 GitHub；国内可填镜像前缀提升成功率');

        o = s.option(form.Button, '_download_core', _(''));
        o.inputtitle = _('下载内核');
        o.inputstyle = 'apply';
        o.onclick = function () {
            return m.save().then(() => clash.downloadCore()).then(() => {
                L.ui.addNotification(null, E('p', _('下载任务已启动，请稍后刷新查看结果')));
            }).catch(function(e) { L.ui.addNotification(null, E('p', '操作失败: ' + (e.message || e))); });
        };

        /* ─── GeoIP / GeoSite ─── */
        s = m.section(form.NamedSection, 'config', 'clash', _('GeoIP / GeoSite 数据库'));
        s.description = _('Mihomo 使用 GeoIP/GeoSite 数据库进行规则匹配，建议定期更新');
        s.anonymous = false;

        o = s.option(form.Flag, 'auto_update_geoip', _('自动更新'));

        o = s.option(form.ListValue, 'auto_update_geoip_time', _('更新时间（每天几点）'));
        for (let i = 0; i <= 23; i++) o.value(String(i), i + ':00');
        o.default = '3';
        o.depends('auto_update_geoip', '1');

        o = s.option(form.Value, 'geoip_update_interval', _('更新周期（天）'));
        o.datatype = 'uinteger';
        o.default = '7';
        o.depends('auto_update_geoip', '1');

        o = s.option(form.ListValue, 'geoip_source', _('数据来源'));
        o.value('2', '默认简化源（推荐）');
        o.value('3', 'OpenClash 社区源');
        o.value('1', 'MaxMind 官方');
        o.value('4', '自定义订阅');
        o.default = '2';

        o = s.option(form.ListValue, 'geoip_format', _('GeoIP 格式'));
        o.value('mmdb', 'MMDB（推荐）');
        o.value('dat', 'DAT');
        o.default = 'mmdb';

        o = s.option(form.ListValue, 'geodata_loader', _('加载模式'));
        o.value('standard', '标准');
        o.value('memconservative', '节省内存');
        o.default = 'standard';
        o.description = _('内存受限设备可选"节省内存"');

        o = s.option(form.Value, 'license_key', _('MaxMind 授权密钥'));
        o.rmempty = true;
        o.depends('geoip_source', '1');

        o = s.option(form.Value, 'geoip_mmdb_url', _('GeoIP（MMDB）订阅链接'));
        o.rmempty = true;
        o.placeholder = 'https://raw.githubusercontent.com/alecthw/mmdb_china_ip_list/release/Country.mmdb';
        o.depends('geoip_source', '2');
        o.depends('geoip_source', '4');

        o = s.option(form.Value, 'geosite_url', _('GeoSite 订阅链接'));
        o.rmempty = true;
        o.placeholder = 'https://github.com/MetaCubeX/meta-rules-dat/releases/download/latest/geosite.dat';
        o.depends('geoip_source', '2');
        o.depends('geoip_source', '3');
        o.depends('geoip_source', '4');

        o = s.option(form.Value, 'geoip_dat_url', _('GeoIP（DAT）订阅链接'));
        o.rmempty = true;
        o.placeholder = 'https://github.com/MetaCubeX/meta-rules-dat/releases/download/latest/geoip.dat';
        o.depends('geoip_source', '2');
        o.depends('geoip_source', '3');
        o.depends('geoip_source', '4');

        o = s.option(form.Button, '_update_geoip', _(''));
        o.inputtitle = _('立即更新 GeoIP');
        o.inputstyle = 'apply';
        o.onclick = function () {
            return m.save().then(() => clash.updateGeoip()).then(() => {
                L.ui.addNotification(null, E('p', _('GeoIP 更新任务已启动')));
            }).catch(function(e) { L.ui.addNotification(null, E('p', '操作失败: ' + (e.message || e))); });
        };

        /* ─── 绕过 ─── */
        s = m.section(form.NamedSection, 'config', 'clash', _('绕过'));
        s.anonymous = false;

        o = s.option(form.Flag, 'bypass_china', _('绕过中国大陆 IP'));
        o.default = '0';
        o.description = _('目标为中国大陆 IP 时直连，不经过代理');

        o = s.option(form.Value, 'china_ip_url', _('大陆 IPv4 段更新 URL'));
        o.placeholder = 'https://ispip.clang.cn/all_cn.txt';
        o.default     = 'https://ispip.clang.cn/all_cn.txt';
        o.rmempty     = true;
        o.depends('bypass_china', '1');

        o = s.option(form.Value, 'china_ipv6_url', _('大陆 IPv6 段更新 URL'));
        o.placeholder = 'https://ispip.clang.cn/all_cn_ipv6.txt';
        o.default     = 'https://ispip.clang.cn/all_cn_ipv6.txt';
        o.rmempty     = true;
        o.depends('bypass_china', '1');

        o = s.option(form.Button, '_update_china_ip', _(''));
        o.inputtitle = _('更新大陆白名单');
        o.inputstyle = 'apply';
        o.depends('bypass_china', '1');
        o.onclick = function () {
            return m.save().then(() => clash.updateChinaIp()).then(() => {
                L.ui.addNotification(null, E('p', _('大陆白名单更新任务已启动，稍后可在更新日志中查看进度')));
            }).catch(function(e) { L.ui.addNotification(null, E('p', '操作失败: ' + (e.message || e))); });
        };

        o = s.option(form.Value, 'proxy_tcp_dport', _('要代理的 TCP 目标端口'));
        o.optional    = true;
        o.placeholder = _('全部端口');
        o.description = _('仅代理指定 TCP 端口，留空表示全部；可填多个空格分隔，如：80 443 8080');

        o = s.option(form.Value, 'proxy_udp_dport', _('要代理的 UDP 目标端口'));
        o.optional    = true;
        o.placeholder = _('全部端口');
        o.description = _('仅代理指定 UDP 端口，留空表示全部；可填多个空格分隔，如：443 8443');

        o = s.option(form.DynamicList, 'bypass_dscp', _('绕过 DSCP'));
        o.datatype = 'range(0, 63)';
        o.rmempty  = true;
        o.description = _('此 DSCP 标记的流量不走代理，范围 0–63');

        o = s.option(form.DynamicList, 'bypass_fwmark', _('绕过 FWMark'));
        o.rmempty  = true;
        o.description = _('此防火墙标记的流量不走代理');

        /* ─── 局域网访问控制 ─── */
        s = m.section(form.NamedSection, 'config', 'clash', _('局域网访问控制'));
        s.description = _('控制哪些局域网设备走代理（按来源 IP 过滤）');
        s.anonymous = false;

        o = s.option(form.ListValue, 'access_control', _('控制模式'));
        o.value('0', '关闭（所有设备走代理）');
        o.value('1', '白名单（仅列表中的设备走代理）');
        o.value('2', '黑名单（列表中的设备不走代理）');
        o.default = '0';

        o = s.option(form.DynamicList, 'proxy_lan_ips', _('白名单设备 IP'));
        o.datatype = 'ip4addr';
        o.rmempty  = true;
        o.retain   = true;
        o.description = _('支持 CIDR，如 192.168.1.100 或 192.168.2.0/24');
        o.depends('access_control', '1');

        o = s.option(form.DynamicList, 'reject_lan_ips', _('黑名单设备 IP'));
        o.datatype = 'ip4addr';
        o.rmempty  = true;
        o.retain   = true;
        o.description = _('支持 CIDR，如 192.168.1.200 或 192.168.3.0/24');
        o.depends('access_control', '2');

        /* ─── 自动化任务 ─── */
        s = m.section(form.NamedSection, 'config', 'clash', _('自动化任务'));
        s.anonymous = false;

        o = s.option(form.Flag, 'auto_update', _('自动更新订阅'));
        o.description = _('定时拉取当前使用的订阅配置');

        o = s.option(form.ListValue, 'auto_update_time', _('更新频率'));
        o.value('1',  '每 1 天');
        o.value('7',  '每 7 天');
        o.value('10', '每 10 天');
        o.value('30', '每 30 天');
        o.default = '7';
        o.depends('auto_update', '1');

        o = s.option(form.Flag, 'auto_clear_log', _('自动清理日志'));

        o = s.option(form.ListValue, 'clear_time', _('清理频率'));
        o.value('1',  '每 1 天');
        o.value('7',  '每 7 天');
        o.value('10', '每 10 天');
        o.value('30', '每 30 天');
        o.default = '7';
        o.depends('auto_clear_log', '1');

        return m.render().then(function (systemNode) {
            /* ─── 日志 Tab 工具函数 ─── */
            function mkLogPanel(initialContent, readFn, clearFn) {
                function processLog(raw) {
                    if (!raw) return '';
                    return raw.trim().split('\n').reverse().join('\n');
                }

                let ta = E('textarea', {
                    style: 'width:100%;height:50vh;font-family:monospace;font-size:13px;padding:8px;resize:vertical;box-sizing:border-box;border:1px solid #d0d0d0;border-radius:4px;background:#fafafa;color:#333;'
                }, [processLog(initialContent)]);

                let clearBtn = E('button', {
                    class: 'btn cbi-button cbi-button-negative',
                    onclick: function () { ta.value = ''; return clearFn().catch(function(e) { L.ui.addNotification(null, E('p', '操作失败: ' + (e.message || e))); }); }
                }, [_('清空日志')]);

                let scrollBtn = E('button', {
                    class: 'btn cbi-button',
                    style: 'margin-left:8px',
                    onclick: function () { ta.scrollTop = 0; }
                }, [_('回到顶部')]);

                let panel = E('div', { style: 'padding-top:12px' }, [
                    E('div', { style: 'margin-bottom:10px' }, [clearBtn, scrollBtn]),
                    ta
                ]);

                poll.add(function () {
                    return readFn().then(function (c) { ta.value = processLog(c); }).catch(function() {});
                }, 5);

                return panel;
            }

            let panels = {
                run:    mkLogPanel(logContent,    () => clash.readLog(),       () => clash.clearLog()),
                update: mkLogPanel(updateContent, () => clash.readUpdateLog(), () => clash.clearUpdateLog()),
                geoip:  mkLogPanel(geoipContent,  () => clash.readGeoipLog(),  () => clash.clearGeoipLog())
            };

            /* ─── 子 Tab 样式（圆角灰底，激活蓝色下划线）─── */
            let TAB_STYLE_BASE   = 'cursor:pointer;padding:8px 18px;margin-right:6px;border-radius:6px 6px 0 0;font-size:14px;border:none;background:#e8eaed;color:#555;border-bottom:3px solid transparent;';
            let TAB_STYLE_ACTIVE = 'cursor:pointer;padding:8px 18px;margin-right:6px;border-radius:6px 6px 0 0;font-size:14px;border:none;background:#e8eaed;color:#4a76d4;border-bottom:3px solid #4a76d4;font-weight:600;';

            function mkBtn(label, active) {
                return E('button', { type: 'button', style: active ? TAB_STYLE_ACTIVE : TAB_STYLE_BASE }, [label]);
            }

            /* ─── 三级子 Tab ─── */
            let subTabs = [
                { key: 'run',    label: _('运行日志') },
                { key: 'update', label: _('更新日志') },
                { key: 'geoip',  label: _('GeoIP 日志') }
            ];

            let subBtns = {};
            let subBar = E('div', { style: 'border-bottom:2px solid #ddd;margin-bottom:12px;padding-top:4px' },
                subTabs.map(function (t) {
                    let b = mkBtn(t.label, t.key === 'run');
                    subBtns[t.key] = b;
                    b.addEventListener('click', function () { switchSub(t.key); });
                    return b;
                })
            );

            function switchSub(key) {
                subTabs.forEach(function (t) {
                    subBtns[t.key].style.cssText = t.key === key ? TAB_STYLE_ACTIVE : TAB_STYLE_BASE;
                    panels[t.key].style.display = t.key === key ? '' : 'none';
                });
            }
            switchSub('run');

            let logSection = E('div', {}, [subBar, panels.run, panels.update, panels.geoip]);

            /* ─── 顶层 Tab（系统设置 / 系统日志）─── */
            let topBtns = {};
            let topDefs = [
                { key: 'system', label: _('系统设置') },
                { key: 'log',    label: _('系统日志') }
            ];
            let topBar = E('div', { style: 'border-bottom:2px solid #ddd;margin-bottom:16px;padding-top:4px' },
                topDefs.map(function (t) {
                    let b = mkBtn(t.label, t.key === 'system');
                    topBtns[t.key] = b;
                    b.addEventListener('click', function () { switchTop(t.key); });
                    return b;
                })
            );

            function switchTop(key) {
                topDefs.forEach(function (t) {
                    topBtns[t.key].style.cssText = t.key === key ? TAB_STYLE_ACTIVE : TAB_STYLE_BASE;
                });
                systemNode.style.display = key === 'system' ? '' : 'none';
                logSection.style.display = key === 'log'    ? '' : 'none';
            }
            switchTop('system');

            return E('div', {}, [
                E('h2', { style: 'margin-bottom:12px' }, [_('系统设置')]),
                topBar,
                systemNode,
                logSection
            ]);
        });
    },

    handleSaveApply: function (ev) {
        return this.handleSave(ev).then(() => clash.restart()).catch(function(e) { L.ui.addNotification(null, E('p', '保存失败: ' + (e.message || e))); });
    }
});

File: luci-app-clashoo/luasrc/model/cbi/clash/system.lua
local NXFS = require "nixio.fs"
local SYS  = require "luci.sys"
local HTTP = require "luci.http"
local DISP = require "luci.dispatcher"
local UTIL = require "luci.util"
local uci  = require("luci.model.uci").cursor()
local fs   = require "luci.clash"

local font_green = [[<font color="green">]]
local font_off   = [[</font>]]
local bold_on    = [[<strong>]]
local bold_off   = [[</strong>]]

-- ── 第一块：内核下载 ──────────────────────────────────────────────
local cpu_model = SYS.exec(
	"opkg status libc 2>/dev/null | grep 'Architecture' | awk -F ': ' '{print $2}' 2>/dev/null"
)

ku = Map("clash")
ku.pageaction = false
s = ku:section(TypedSection, "clash", "内核下载")
s.anonymous = true
s.addremove  = false
s.description = "从 GitHub 下载 Mihomo 内核二进制文件"

o = s:option(ListValue, "dcore", "版本类型")
o.default = "3"
o:value("2", "mihomo（稳定版）")
o:value("3", "Alpha（预发布版）")
o.description = "稳定版适合日常使用，Alpha 版含最新功能但可能有不稳定风险"

o = s:option(ListValue, "download_core", "CPU 架构")
o.default     = "x86_64"
o.description = "当前设备：" .. font_green .. bold_on .. cpu_model:gsub("%s+$", "") .. bold_off .. font_off
o:value("aarch64_cortex-a53")
o:value("aarch64_generic")
o:value("arm_cortex-a7_neon-vfpv4")
o:value("mipsel_24kc")
o:value("mips_24kc")
o:value("x86_64")
o:value("riscv64")

o = s:option(Value, "core_mirror_prefix", "下载镜像前缀（可选）")
o.rmempty     = true
o.placeholder = "https://gh-proxy.com/"
o.description = "留空直连 GitHub；国内可填镜像前缀提升成功率"

o = s:option(Button, "down_core", "")
o.inputtitle = "保存配置"
o.inputstyle = "apply"
o.write = function()
	ku.uci:commit("clash")
end

o = s:option(Button, "download", "")
o.template = "clash/core_check"

-- ── 第二块：GeoIP 数据库 ──────────────────────────────────────────
local function geoip_update_time()
	local paths = {
		"/etc/clash/Country.mmdb",
		"/usr/share/clash/geoip/Country.mmdb",
		"/usr/share/mihomo/Country.mmdb",
	}
	for _, p in ipairs(paths) do
		if NXFS.access(p) then
			return SYS.exec("ls -l --full-time " .. p .. " | awk '{print $6, $7}'")
		end
	end
	return "-"
end

gm = Map("clash")
gm.pageaction = false
gs = gm:section(TypedSection, "clash", "GeoIP / GeoSite 数据库")
gs.anonymous = true
gs.addremove  = false
gs.description = "Mihomo 使用 GeoIP/GeoSite 数据库进行规则匹配，建议定期更新"

o = gs:option(Flag, "auto_update_geoip", "自动更新")
o.description = "开启后按指定周期自动更新数据库"

o = gs:option(ListValue, "auto_update_geoip_time", "更新时间（每天几点）")
for t = 0, 23 do o:value(t, t .. ":00") end
o.default = 3
o:depends("auto_update_geoip", "1")

o = gs:option(Value, "geoip_update_interval", "更新周期（天）")
o.datatype    = "uinteger"
o.default     = 7
o.rmempty     = false
o.description = "例如 7 表示每隔 7 天更新"
o:depends("auto_update_geoip", "1")

o = gs:option(ListValue, "geoip_source", "数据来源")
o:value("2", "默认简化源（推荐）")
o:value("3", "OpenClash 社区源")
o:value("1", "MaxMind 官方")
o:value("4", "自定义订阅")
o.default = "2"

o = gs:option(ListValue, "geoip_format", "GeoIP 格式")
o:value("mmdb", "MMDB（推荐）")
o:value("dat",  "DAT")
o.default     = "mmdb"
o.description = "一般无需更改"

o = gs:option(ListValue, "geodata_loader", "加载模式")
o:value("standard",       "标准")
o:value("memconservative", "节省内存")
o.default     = "standard"
o.description = [[内存受限设备可选"节省内存"]]

o = gs:option(Value, "license_key", "MaxMind 授权密钥")
o.description = "仅 MaxMind 官方来源需要，申请地址：maxmind.com"
o.rmempty     = true
o:depends("geoip_source", "1")

o = gs:option(Value, "geoip_mmdb_url", "GeoIP（MMDB）订阅链接")
o.rmempty     = true
o.placeholder = "https://raw.githubusercontent.com/alecthw/mmdb_china_ip_list/release/Country.mmdb"
o:depends("geoip_source", "2")
o:depends("geoip_source", "4")

o = gs:option(Value, "geosite_url", "GeoSite 订阅链接")
o.rmempty     = true
o.placeholder = "https://github.com/MetaCubeX/meta-rules-dat/releases/download/latest/geosite.dat"
o:depends("geoip_source", "2")
o:depends("geoip_source", "3")
o:depends("geoip_source", "4")

o = gs:option(Value, "geoip_dat_url", "GeoIP（DAT）订阅链接")
o.rmempty     = true
o.placeholder = "https://github.com/MetaCubeX/meta-rules-dat/releases/download/latest/geoip.dat"
o:depends("geoip_source", "2")
o:depends("geoip_source", "3")
o:depends("geoip_source", "4")

o = gs:option(Value, "geoip_asn_url", "GeoIP（ASN）订阅链接")
o.rmempty     = true
o.placeholder = "https://github.com/MetaCubeX/meta-rules-dat/releases/download/latest/GeoLite2-ASN.mmdb"
o:depends("geoip_source", "2")
o:depends("geoip_source", "4")

o = gs:option(DummyValue, "_geoip_last_update", "上次更新时间")
o.rawhtml  = true
o.cfgvalue = function()
	local t = geoip_update_time():gsub("%s+$", "")
	return "<strong>" .. (t ~= "" and t or "-") .. "</strong>"
end

o = gs:option(Button, "update_geoip", "")
o.inputtitle = "保存配置"
o.inputstyle = "apply"
o.write = function()
	gm.uci:commit("clash")
end

o = gs:option(Button, "download", "")
o.template = "clash/geoip"


-- ── 第二块半：绕过 ────────────────────────────────────────────────
bm = Map("clash")
bm.pageaction = false
bs = bm:section(TypedSection, "clash", "绕过")
bs.anonymous = true
bs.addremove  = false

-- 绕过目标 IP（不代理这些目的地）
o = bs:option(Flag, "bypass_china", "绕过中国大陆 IP")
o.rmempty     = false
o.default     = "0"
o.description = "目标为中国大陆 IP 时直连，不经过代理（需要 /usr/share/clash/china_ip.txt）"

o = bs:option(Button, "_save_bypass", "")
o.inputtitle = "保存"
o.inputstyle = "apply"
o.write = function()
	bm.uci:commit("clash")
end

-- ── 第二块三：局域网访问控制 ──────────────────────────────────────
am2 = Map("clash")
am2.pageaction = false
as2 = am2:section(TypedSection, "clash", "局域网访问控制")
as2.anonymous = true
as2.addremove  = false
as2.description = "控制哪些局域网设备走代理（按来源 IP 过滤）"

o = as2:option(ListValue, "access_control", "控制模式")
o:value("0", "关闭（所有设备走代理）")
o:value("1", "白名单（仅列表中的设备走代理）")
o:value("2", "黑名单（列表中的设备不走代理）")
o.default     = "0"

o = as2:option(DynamicList, "proxy_lan_ips", "白名单设备 IP")
o.datatype    = "ip4addr"
o.rmempty     = true
o.placeholder = "如：192.168.1.100 或 192.168.2.0/24"
o.description = "手动输入要走代理的设备 IP，支持 CIDR"
o:depends("access_control", "1")

o = as2:option(DynamicList, "reject_lan_ips", "黑名单设备 IP")
o.datatype    = "ip4addr"
o.rmempty     = true
o.placeholder = "如：192.168.1.200 或 192.168.3.0/24"
o.description = "手动输入不走代理的设备 IP，支持 CIDR"
o:depends("access_control", "2")

o = as2:option(Button, "_save_ac", "")
o.inputtitle = "保存"
o.inputstyle = "apply"
o.write = function()
	am2.uci:commit("clash")
end

-- ── 第三块：自动化 ────────────────────────────────────────────────
am = Map("clash")
am.pageaction = false
as = am:section(TypedSection, "clash", "自动化任务")
as.anonymous = true
as.addremove  = false

o = as:option(Flag, "auto_update", "自动更新订阅")
o.description = "定时拉取当前使用的订阅配置"

o = as:option(ListValue, "auto_update_time", "更新频率")
o:value("1",  "每小时")
o:value("6",  "每 6 小时")
o:value("12", "每 12 小时")
o:value("24", "每 24 小时")
o.description = "仅更新当前正在使用的配置"

o = as:option(Flag, "auto_clear_log", "自动清理日志")
o.description = "定时清空运行日志文件"

o = as:option(ListValue, "clear_time", "清理频率")
o:value("1",  "每小时")
o:value("6",  "每 6 小时")
o:value("12", "每 12 小时")
o:value("24", "每 24 小时")
o:depends("auto_clear_log", "1")

o = as:option(Button, "_save_auto", "")
o.inputtitle = "保存"
o.inputstyle = "apply"
o.write = function()
	am.uci:commit("clash")
end

-- ── 第四块：运行日志 ──────────────────────────────────────────────
lm = Map("clash")
lm.pageaction = false
ls = lm:section(TypedSection, "clash", "运行日志")
ls.anonymous = true
ls.addremove  = false

log = ls:option(TextValue, "clog")
log.template = "clash/status_log"

o = ls:option(Button, "Download", "")
o.inputtitle = "下载日志文件"
o.inputstyle = "apply"
o.write = function()
	local sPath = "/usr/share/clash/clash.txt"
	local sFile = NXFS.basename(sPath)
	local fd    = nixio.open(sPath, "r")
	if not fd then return end
	HTTP.header("Content-Disposition", 'attachment; filename="' .. sFile .. '"')
	HTTP.prepare_content("application/octet-stream")
	while true do
		local block = fd:read(nixio.const.buffersize)
		if not block or #block == 0 then break end
		HTTP.write(block)
	end
	fd:close()
	HTTP.close()
end

return ku, gm, bm, am2, am, lm
